pkg/registry/file/vulnsummarystorage.go

```diff
@@ -0,0 +1,240 @@
+	"github.com/spf13/afero"
+	"go.opentelemetry.io/otel"
+	"go.opentelemetry.io/otel/attribute"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/watch"
+	"k8s.io/apiserver/pkg/storage"
+)
+
+// VulnSummaryStorageImpl offers a common interface for object marshaling/unmarshaling operations and
+// hides all the storage-related operations behind it.
+type VulnSummaryStorageImpl struct {
+	appFs     afero.Fs
+	root      string
+	versioner storage.Versioner
+}
+
+var _ storage.Interface = &StorageImpl{}
+
+func NewVulnSummaryStorageImpl(appFs afero.Fs, root string) storage.Interface {
+	return &VulnSummaryStorageImpl{
+		appFs:     appFs,
+		root:      root,
+		versioner: storage.APIObjectVersioner{},
+	}
+}
+
+// Versioner Returns Versioner associated with this interface.
+func (s *VulnSummaryStorageImpl) Versioner() storage.Versioner {
+	return s.versioner
+}
+
+func (s *VulnSummaryStorageImpl) Create(ctx context.Context, key string, obj, out runtime.Object, _ uint64) error {
+	_, span := otel.Tracer("").Start(ctx, "VulnSummaryStorageImpl.Create")
+	span.SetAttributes(attribute.String("key", key))
+	defer span.End()
+	return storage.NewMethodNotImplementError(key, "")
+}
+
+func (s *VulnSummaryStorageImpl) Delete(ctx context.Context, key string, out runtime.Object, _ *storage.Preconditions, _ storage.ValidateObjectFunc, _ runtime.Object) error {
+	_, span := otel.Tracer("").Start(ctx, "VulnSummaryStorageImpl.Delete")
+	span.SetAttributes(attribute.String("key", key))
+	defer span.End()
+	return storage.NewMethodNotImplementError(key, "")
+}
+
+// add uni test
+func (s *VulnSummaryStorageImpl) Watch(ctx context.Context, key string, _ storage.ListOptions) (watch.Interface, error) {
+	_, span := otel.Tracer("").Start(ctx, "VulnSummaryStorageImpl.Watch")
+	span.SetAttributes(attribute.String("key", key))
+	defer span.End()
+	return nil, storage.NewMethodNotImplementError(key, "")
+}
+
+// add uni test
+func (s *VulnSummaryStorageImpl) getOneVulnManifestSummary(ctx context.Context, path string) (*softwarecomposition.VulnerabilityManifestSummary, error) {
+	var vulnManifestSummary softwarecomposition.VulnerabilityManifestSummary
+
+	b, err := afero.ReadFile(s.appFs, path)
+	if err != nil {
+		if errors.Is(err, afero.ErrFileNotFound) {
+			return nil, storage.NewKeyNotFoundError(path, 0)
+		}
+		logger.L().Ctx(ctx).Error("read file failed", helpers.Error(err), helpers.String("key", path))
+		return nil, err
+	}
+	err = json.Unmarshal(b, &vulnManifestSummary)
+	if err != nil {
+		logger.L().Ctx(ctx).Error("json unmarshal failed", helpers.Error(err), helpers.String("key", path))
+		return nil, err
+	}
+	if _, exist := vulnManifestSummary.GetLabels()["kubescape.io/workload-namespace"]; !exist {
+		return nil, storage.NewKeyNotFoundError(path, 0)
+	}
+	return &vulnManifestSummary, nil
+}
+
+// add uni test
+func initNamespaceVulnSummary(key string) softwarecomposition.VulnerabilitySummary {
+	namespace := filepath.Base(key)
```

Comment:
issue: namespace should not be computed inside the function, it should be a parameter

Essentially, this function is a constructor for VulnerabilitySummaries. That means, it should create an empty struct and populate it with necessary default values. It is not the job of the constructor to compute these values from whatever noise comes into it.

Instead, if we review what happens on a high level inside the storage, the flow is as follows:
- A user makes a request to K8s API
- K8s API forwards this request to our Storage APIServer
- Storage APIServer routes this to appropriate `REST` objects
- The `REST` object passes the `path` value to the `Storage` objects
- The `Storage` object starts handling the requested operation.
- The inner objects handle things based on some business logic we define.

Now, this business logic cares about creating a Vulnerability Summary for some scope, be it namespace or cluster. It does not care about the entire path parameter. And it should not even know about the concept of an element path. All it has to know is the name of the scope that it needs to create a Vulnerability Summary object for. 

Which is why this function should take a `namespace` string instead of computing it on its own. The caller should supply the namespace.